### PR TITLE
AA-182: Detect violation of rule "4.1 Must not use value" in the bundler

### PR DESCRIPTION
**Pull request by forshtat**
Validates fixes in https://github.com/eth-infinitism/bundler/pull/136

### CODE REVIEW

File: tests/bundle/test_storage_rules.py [hunk 1]
```diff
@@ -336,6 +348,27 @@ def build_userop_for_factory(w3, entrypoint_contract, factory_contract, rule):
...
+        assert_error,
+    ),
+    StorageTestCase(
+        "eth_value_transfer",
+        UNSTAKED,
+        SENDER,
+        with_ether_value(build_userop_for_sender),
+        assert_error,
+    ),
+    StorageTestCase(
```
Comment (drortirosh) [inline tests/bundle/test_storage_rules.py:365]: must make sure we test account deposit with both `entrypoint.call{value}` and `entrypoint.depositTo{value}()`
(one of them is the default account impl, the other should have explicit test

Comment (forshtat) [reply tests/bundle/test_storage_rules.py:353]: Added.

File: tests/bundle/test_storage_rules.py [hunk 2]
```diff
@@ -46,6 +46,18 @@ def deploy_staked_factory(w3, entrypoint_contract):
     )
 
 
+def with_ether_value(build_userop_func):
```
[the remark this answers is not in the document]
Comment (forshtat) [reply tests/bundle/test_storage_rules.py]: Replaced `depositTo` with `sendTransaction` in Utils.py. This did require adding some `receive() payable` functions in other solidity contracts.

File: tests/contracts/TestFakeWalletToken.sol [hunk 3]
```diff
@@ -41,6 +42,10 @@ contract TestFakeWalletToken is IAccount {
         return 0;
     }
 
+    receive() external payable {
```
Comment (drortirosh) [inline tests/contracts/TestFakeWalletToken.sol:45]: this is not good to pass anything to entryPoint.

Comment (drortirosh) [reply tests/contracts/TestFakeWalletToken.sol:45]: instead add to validateUserOp
```
 if (missingWalletFunds>0) {
   msg.sender.call{value:missingWalletFunds}("");
}
```

**Review by drortirosh — APPROVED** [review]

**Pull request merged**